src/OpenTelemetry.Abstractions/Trace/Tracing.cs

```diff
@@ -0,0 +1,81 @@
+﻿// <copyright file="Tracing.cs" company="OpenTelemetry Authors">
+// Copyright 2018, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+namespace OpenTelemetry.Trace
+{
+    using System.Threading;
+    using System.Threading.Tasks;
+    using OpenTelemetry.Context;
+    using OpenTelemetry.Context.Propagation;
+    using OpenTelemetry.Trace.Export;
+
+    /// <summary>
+    /// Class that manages a global instance of the <see cref="Tracer"/>.
+    /// </summary>
+    public static class Tracing
+    {
+        private static readonly Proxy ProxyInstance = new Proxy();
+
+        /// <summary>
+        /// Gets the tracer to record spans.
+        /// </summary>
+        public static ITracer Tracer { get; } = ProxyInstance;
+
+        /// <summary>
+        /// Gets the exporter to use to upload spans.
+        /// </summary>
+        public static ISpanExporter SpanExporter { get; } = ProxyInstance;
+
+        public static void Init(ITracer tracer, ISpanExporter spanExporter)
+        {
+            ProxyInstance.ActualTracer = tracer ?? NoopTracer.Instance;
+            ProxyInstance.ActualSpanExporter = spanExporter ?? NoopSpanExporter.Instance;
+        }
+
+        private class Proxy : ITracer, ISpanExporter
```

Comment:
This seems to be very had to maintain as we move forward. 

Can Noop tracer be this proxy and then it seems we can avoid making everything in default tracing ?

Reply:
I'm sorry I don't follow. You'd like `NoopTracer` to be the proxy meaning we never remove that instance from `Tracing.Tracer`? So We'd modify `NoopTracer` to take an actual instance and no longer be `noop`?

Reply:
I'm not convinced of having an instance that doesn't change (aka: `Proxy`) due to [the issue I mentioned here](https://github.com/open-telemetry/opentelemetry-dotnet/pull/209#issuecomment-533875444). But if the goal is to not replace the reference held via `Tracing.Tracer` anything that is internal to `OpenTelemetry.Abstractions` shouldn't be too hard to maintain as we can change without breaking changes. So if the inner class isn't the way to go, we could move to `internal` classes (two separate ones I assume would be best).

Reply:
With changes I'm proposing here: https://github.com/open-telemetry/opentelemetry-dotnet/blob/3751a863b9ca0ccac0f6d81f83fa81832347afad/src/OpenTelemetry/Trace/Tracing.cs we only need ProxyTracer (no exporter methods) for good. I.e. you current approach seems to be fine with me.